indexing-service/src/main/java/org/apache/druid/indexing/common/config/TaskConfig.java

```diff
@@ -175,6 +211,14 @@ public TaskConfig(
     }
     log.debug("Batch processing mode:[%s]", this.batchProcessingMode);
     this.storeEmptyColumns = storeEmptyColumns == null ? DEFAULT_STORE_EMPTY_COLUMNS : storeEmptyColumns;
+
+    if (baseTaskDirLocations == null || baseTaskDirLocations.size() == 0) {
```

Comment:
Can use `org.apache.druid.utils.CollectionUtils.isEmpty` to replace these two conditions